src/Riok.Mapperly/Descriptors/Mappings/TypeMappingBuildContext.cs

```diff
@@ -30,15 +32,17 @@ SyntaxFactoryHelper syntaxFactory
         SyntaxFactory = syntaxFactory;
     }
 
-    public UniqueNameBuilder NameBuilder { get; }
+    public UniqueNameBuilder NameBuilder { get; private init; }
 
-    public ExpressionSyntax Source { get; }
+    public ExpressionSyntax Source { get; private init; }
 
-    public ExpressionSyntax? ReferenceHandler { get; }
+    public IReadOnlyList<IMappableMember> TrimSourcePath { get; private init; } = ImmutableList<IMappableMember>.Empty;
```

Comment:
I can work a bit on the documentation side.

This property should contain `A, B` when a full path is `A.B?.C` so it can replace `A.B` with the newly introduced guard variable.